src/datasource.ts

```diff
@@ -92,7 +92,7 @@ export class GenericDatasource {
     });
   }
 
-  findMetricsQuery(type: string, query: string) {
+  findMetricsQuery(query: string, type: string) {
```

Comment:
also, the `type`'s type should be enum of `timeseries` and `table` (?)

Reply:
Is there such a type anywhere? If not leave as is as we're only passing it through and more validation won't really help us here.

Reply:
AFAIK not yet, I guess Grafana will add it later

Reply:
Ah, it's our thing actually https://github.com/simPod/grafana-json-datasource/commit/2c49ff836c36954c1a0edcb640facc1e36d59901

Reply:
Not sure if you want me to change anything here? As not an upstream type I'd leave this as is.

Reply:
I wanted to create the enum but you can skip it and I'll do it later